### PR TITLE
Run scan endpoint OXO spec

### PR DESCRIPTION
**Pull request by elyousfi5**

### CODE REVIEW

File: src/ostorlab/serve_app/oxo.py [hunk 1]
```diff
@@ -113,5 +122,96 @@ def mutate(
...
+                    f"Agent group definition YAML parse error:: {e}"
+                )
+        else:
+            raise graphql.GraphQLError("Missing agent list or agent group definition.")
+
+        asset_group = None
+        if assets is not None:
+            try:
+                asset_group = definitions.AssetsDefinition.from_yaml(assets)
+                pass
```
Comment (benyissa) [inline src/ostorlab/serve_app/oxo.py:174]:
```suggestion
```

File: src/ostorlab/runtimes/local/runtime.py [hunk 2]
```diff
@@ -222,22 +222,30 @@ def scan(
             self._update_scan_progress("IN_PROGRESS")
             console.success("Scan created successfully")
         except AgentNotHealthy:
-            console.error("Agent not starting")
+            message = "Agent not starting"
+            console.error(message)
```
Comment (3asm) [inline src/ostorlab/runtimes/local/runtime.py:226]: Why!!!

File: src/ostorlab/runtimes/local/runtime.py [hunk 3]
```diff
@@ -222,22 +222,30 @@ def scan(
...
             self.stop(self._scan_db.id)
             self._update_scan_progress("ERROR")
             self.stop(str(self._scan_db.id))
+            return message
```
Comment (3asm) [inline src/ostorlab/runtimes/local/runtime.py:230]: API returning a message to detente status is a BAAADD practice.

**Review by 3asm — CHANGES_REQUESTED** [review]

**Comment by 3asm** [comment]
Ping @elyousfi5 what is the status of this PR?

**Comment by elyousfi5** [comment]
> Ping @elyousfi5 what is the status of this PR?

Depends on #679

**Comment by 3asm** [comment]
Ping on PR @elyousfi5

File: src/ostorlab/cli/scan/run/run.py [hunk 4]
```diff
@@ -171,11 +171,14 @@ def run(
...
-                assets=asset_group if asset_group is None else asset_group.targets,
-            )
+            try:
+                runtime_instance.scan(
+                    title=ctx.obj["title"],
+                    agent_group_definition=ctx.obj["agent_group_definition"],
+                    assets=asset_group.targets if asset_group is not None else None,
+                )
+            except exceptions.OstorlabError as e:
+                logger.error(f"Error running scan: {e}")
```
Comment (oussamaessaji) [inline src/ostorlab/cli/scan/run/run.py:181]:
```suggestion
                logger.error("Error running scan: %s", e)
```

File: src/ostorlab/cli/scan/run/run.py [hunk 5]
```diff
@@ -171,11 +171,14 @@ def run(
...
-                assets=asset_group if asset_group is None else asset_group.targets,
-            )
+            try:
+                runtime_instance.scan(
+                    title=ctx.obj["title"],
+                    agent_group_definition=ctx.obj["agent_group_definition"],
+                    assets=asset_group.targets if asset_group is not None else None,
+                )
+            except exceptions.OstorlabError as e:
+                logger.error("Error running scan: %s", e)
```
Comment (3asm) [inline src/ostorlab/cli/scan/run/run.py:181]: Why wrap it, we want ti to fail and trigger a graphql error?

File: src/ostorlab/serve_app/oxo.py [hunk 6]
```diff
@@ -456,6 +473,189 @@ def mutate(
...
+            scan: The scan information.
+
+        Raises:
+            graphql.GraphQLError in case of an error.
+
+        Returns:
+            The scan information.
+        """
+
+        with models.Database() as session:
```
Comment (3asm) [inline src/ostorlab/serve_app/oxo.py:501]: Should be split into separate methods.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: src/ostorlab/serve_app/oxo.py [hunk 7]
```diff
@@ -456,6 +473,189 @@ def mutate(
...
+                        args=[
+                            utils_definitions.Arg.build(
+                                name=arg.name,
+                                type=arg.type,
+                                value=arg.value,
+                                description=arg.description,
+                            )
+                            for arg in session.query(models.AgentArgument)
+                            .filter_by(agent_id=agent.id)
+                            .all()
```
Comment (oussamaessaji) [inline src/ostorlab/serve_app/oxo.py:538]: put it in a var

**Review by benyissa — DISMISSED** [review]
Instead of catching OstorlabError at each type of asset, Why not do it inside the runtime classes as we did for other exceptions?

File: src/ostorlab/runtimes/local/models/models.py [hunk 8]
```diff
@@ -160,7 +160,7 @@ def create(
                 title=title,
                 asset=asset,
                 created_time=datetime.datetime.now(),
-                progress=progress.name,
+                progress=progress,
```
Comment (mohsinenar) [inline src/ostorlab/runtimes/local/models/models.py:163]: Progress is enum right?

**Review by 3asm — APPROVED** [review]

File: src/ostorlab/runtimes/local/models/models.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (elyousfi5) [reply src/ostorlab/runtimes/local/models/models.py:164]: progress is enum
image

**Review by 3asm — APPROVED** [review]

**Comment by 3asm** [comment]
Please resolve conflict and it is good to go.

**Review by ErebusZ — APPROVED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**